src/main/java/com/amazon/ion/impl/IonReaderContinuableCoreBinary.java

```diff
@@ -2473,7 +2473,8 @@ protected final SymbolToken getSystemSymbolToken(Marker marker) {
         if (id == 0) {
             return _Private_Utils.SYMBOL_0;
         }
-        return SystemSymbols_1_1.get((int) id).getToken();
+        SystemSymbols_1_1 systemSymbol = SystemSymbols_1_1.get((int) id);
+        return (systemSymbol == null) ? new SymbolTokenImpl(-1) : systemSymbol.getToken();
```

Comment:
It looks to me like throwing is the right thing here. This is only called in places where the encoding indicates a system symbol is present. It would be invalid Ion for that encoding to occur without a valid system symbol mapping.

Reply:
We probably need to throw an `IonException` here because if we have something that claims to be a system symbol, but the ID is out of range for the system symbol table, it's invalid Ion data.